harness/determined/exec/tensorboard.py

```diff
@@ -26,121 +40,108 @@ def set_s3_region() -> None:
-        *args,
+        *add_args,
     ]
 
-    major, minor = get_tensorboard_version(version)
-
-    if major == "2":
+    # Version dependant args
+    if version.parse(tb_version) >= version.parse("2"):
         tensorboard_args.append("--bind_all")
-        if minor >= "5":
-            tensorboard_args.append("--load_fast=false")
-        if len(logdir.split(",")) > 1:
-            tensorboard_args.append(f"--logdir_spec={logdir}")
-            return tensorboard_args
+    if version.parse(tb_version) >= version.parse("2.5"):
+        tensorboard_args.append("--load_fast=false")
 
-    tensorboard_args.append(f"--logdir={logdir}")
-    return tensorboard_args
+    tensorboard_args.append(f"--logdir={tfevents_dir}")
 
+    return tensorboard_args
 
-def main(args: List[str]) -> int:
-    set_s3_region()
 
+def get_tensorboard_url() -> str:
     task_id = os.environ["DET_TASK_ID"]
     port = os.environ["TENSORBOARD_PORT"]
     tensorboard_addr = f"http://localhost:{port}/proxy/{task_id}"
-    url = f"{tensorboard_addr}/data/plugin/scalars/tags"
-    tensorboard_args = get_tensorboard_args(args)
+    return f"{tensorboard_addr}/data/plugin/scalars/tags"
+
+
+def start_tensorboard(
+    config: Dict[str, Any],
+    tb_version: str,
+    storage_paths: List[str],
+    add_tb_args: List[str],
+) -> int:
+    """Start Tensorboard and look for new files."""
+
+    stop_time = time.time() + MAX_WAIT_TIME
+    triggered = False
 
-    print(f"Running: {tensorboard_args}")
-    p = subprocess.Popen(tensorboard_args)
+    # Create local temporary directory
+    local_dir = os.path.join(tempfile.gettempdir(), "tf_events")
+    os.makedirs(local_dir, mode=0o777, exist_ok=True)
 
-    def still_alive() -> bool:
-        return p.poll() is None
+    # Get fetcher and perform initial fetch
+    fetcher = fetchers.build(config, storage_paths, local_dir)
+    num_fetched_files = fetcher.fetch_new()
 
-    if not wait_for_tensorboard(600, url, still_alive):
-        p.kill()
+    # Build Tensorboard args and launch process.
+    tb_args = get_tensorboard_args(tb_version, local_dir, add_tb_args)
+    logger.debug(f"tensorboard args: {tb_args}")
+    tensorboard_process = subprocess.Popen(tb_args)
 
-    return p.wait()
+    try:
+        while True:
+            ret_code = tensorboard_process.poll()
+            if ret_code is not None:
+                raise RuntimeError(f"Tensorboard process died, exit code({ret_code}).")
+
+            # Check if we have reached a timeout without receiving metrics
+            if num_fetched_files == 0 and time.time() > stop_time:
+                raise RuntimeError("We reached the timeout without receiving metrics.")
+
+            if not triggered and num_fetched_files > 0:
+                print(TENSORBOARD_TRIGGER_READY_MSG)
+                triggered = True
+
+            num_fetched_files += fetcher.fetch_new()
+            time.sleep(FETCH_INTERVAL)
+
+    except Exception as exp:
+        logger.error(str(exp))
```

Comment:
Why even catch the error?  When there is an error you print the error text (but not the stack trace), then you kill the tensorboard and exit with the exit code of the tensorboard process.  If you didn't catch the error, you would instead exit printing an entire stack trace, which seems preferable to me.

I'd much rather see a whole stacktrace than a `NoneType error` or some unhelpful string that could have originated in a variety of different files.

Reply:
I think it is left over code, which doesn't make sense, from the review process. Removed.